ntbs-service/wwwroot/source/Components/AutocompleteSelect.ts

```diff
@@ -0,0 +1,65 @@
+import Vue from "vue";
+var accessible = require("accessible-autocomplete");
+
+const AutocompleteSelect = Vue.extend({
+    props: ["placeholder", "validate"],
+    mounted: function() {
+        var selectElement = this.$el;
+        var selectId = selectElement.id;
+        accessible.enhanceSelectElement({
+            placeholder: this.$props.placeholder,
+            autoselect: false,
+            preserveNullOptions: true,
+            showAllValues: true,
+            selectElement: selectElement,
+            onConfirm: (val: any) => {
+                this.onConfirm(val);
+            }
+        })
+        this.autocompleteElement = document.getElementById(selectId) as HTMLInputElement;
+        this.selectElement = document.getElementById(this.autocompleteElement.id + "-select") as HTMLSelectElement;

```

Comment:
Used `selectElement` symbol twice (pointing to the vanilla un-modified select, and the second time to the modified hidden selectElement). 
Technically they were two different variables, local variable and dynamic property on the closure - but hurts readability.

It might be nice to add state data for autocompleteElement and selectElement. This would be quite nice for debug purposes too (having easy access to the state data either in vanilla devtools console - or with the vue devtools).

Reply:
I tried listening to state changes on the original select element with change, input etc.. none of them get picked up :(

Reply:
When I say state data here, I meant in vue terms:
https://vuejs.org/v2/guide/components.html#Base-Example

Even if we're not adding state for them, I think a variable name change to make a clearer distinction between var.X and this.X would be apt.

Reply:
Sorry I think I misunderstood your question - I just realised there is no need to define a local variable, I can just define this.selectElement straight away (using a ref property) and it will refer to the correct element. Will be clearer once I push my changes